cmd/cmd_util.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright © 2021 The Persistent-Volume-Migrator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	"persistent-volume-migrator/internal/ceph/rbd"
+	"persistent-volume-migrator/internal/kubernetes"
+	logger "persistent-volume-migrator/internal/log"
+
+	v1 "k8s.io/api/core/v1"
+	k8s "k8s.io/client-go/kubernetes"
+)
+
+// migratePVC migrates a list of PVCs to CSI.
+func migratePVC(client *k8s.Clientset, pvcs *[]v1.PersistentVolumeClaim) error {
+	var err error
+	for _, pvc := range *pvcs {
+		pv, err := kubernetes.GetPV(client, pvc.Spec.VolumeName)
+		if err != nil {
+			return fmt.Errorf("failed to get PV object with name %s: %v", pvc.Spec.VolumeName, err)
+		}
+		logger.DefaultLog("PV found %v ", pv)
+		err = kubernetes.UpdateReclaimPolicy(client, pv)
+		if err != nil {
+			return fmt.Errorf("failed to update ReclaimPolicy for PV object %s: %v", pvc.Spec.VolumeName, err)
+		}
+		// Retrive old ceph volume name from PV
+		rbdImageName := kubernetes.GetFlexVolumeName(pv)
```

Comment:
Would this also work for an in-tree rbd volume conversion, or just flex?
```suggestion
		rbdImageName := kubernetes.GetFlexVolumeName(pv)
```

Reply:
Although we are definitely looking forward to supporting both flex and tree, the focus for the first release will be flex and it is of higher priority. After then, with some tweaks and a CI, we will be ready for the migration of intree as well :)